From 1854e1eaa51d2c571c5c8c97e5eddcaa6d632726 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Mon, 24 Apr 2023 15:43:16 +0200
Subject: [PATCH 0304/1695] dts: stm32: populate RISAB nodes

RISABs are default enabled and cover internal RAMs.

Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: Ia1c01908368935d45819be0a6e9b332551816487

squash! dts: stm32: populate RISAB nodes

RISAB protect internal RAMs. Default enable ALL of them.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 583cc03ed2bb1e5e3358bf0c8e2d1e88abf850d1 dts: stm32: populate RISAB nodes]
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: I9d98b837abcc685704d873ddfaed34d6ba3f03cd
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/336351
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/dts/stm32mp251.dtsi | 23 ++++++++++++++++++-----
 1 file changed, 18 insertions(+), 5 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index 933c833d1..729abaecc 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -257,32 +257,45 @@
 		};
 
 		risab1: risab@420f0000 {
+			compatible = "st,stm32mp25-risab";
 			reg = <0x420f0000 0x1000>;
-			status = "disabled";
+			clocks = <&rcc CK_ICN_LS_MCU>;
+			st,mem-map = <0xa000000 0x20000>;
 		};
 
 		risab2: risab@42100000 {
+			compatible = "st,stm32mp25-risab";
 			reg = <0x42100000 0x1000>;
-			status = "disabled";
+			clocks = <&rcc CK_ICN_LS_MCU>;
+			st,mem-map = <0xa020000 0x20000>;
 		};
 
 		risab3: risab@42110000 {
+			compatible = "st,stm32mp25-risab";
 			reg = <0x42110000 0x1000>;
-			status = "disabled";
+			clocks = <&rcc CK_ICN_LS_MCU>;
+			st,mem-map = <0xa040000 0x20000>;
 		};
 
 		risab4: risab@42120000 {
+			compatible = "st,stm32mp25-risab";
 			reg = <0x42120000 0x1000>;
-			status = "disabled";
+			clocks = <&rcc CK_ICN_LS_MCU>;
+			st,mem-map = <0xa060000 0x20000>;
 		};
 
 		risab5: risab@42130000 {
+			compatible = "st,stm32mp25-risab";
 			reg = <0x42130000 0x1000>;
-			status = "disabled";
+			clocks = <&rcc CK_ICN_LS_MCU>;
+			st,mem-map = <0xa080000 0x20000>;
 		};
 
 		risab6: risab@42140000 {
+			compatible = "st,stm32mp25-risab";
 			reg = <0x42140000 0x1000>;
+			clocks = <&rcc CK_ICN_LS_MCU>;
+			st,mem-map = <0xa0a0000 0x20000>;
 			status = "disabled";
 		};
 
-- 
2.47.2

